kinto/core/permission/postgresql/__init__.py

```diff
@@ -63,12 +66,68 @@ class Permission(PermissionBase):
 
     :noindex:
     """  # NOQA
+
+    name = 'permission'
+    schema_version = 2
+    schema_file = os.path.join(HERE, 'schema.sql')
+    migrations_directory = os.path.join(HERE, 'migrations')
+
     def __init__(self, client, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.client = client
 
     def initialize_schema(self, dry_run=False):
-        # Check if user_principals table exists.
+        return self.create_or_migrate_schema(dry_run)
+
+    def _get_installed_version(self):
+        """Return current version of schema or None if not any found.
+
+        Migrations were only added to the permission backend in
+        8.1.2. Before this, the permission backend was only the two
+        tables ``user_principals`` and ``access_control_entries``. The
+        presence of these two tables and absence of a metadata
+        table/permission_schema_version is therefore version 1.
+
+        In version 8.1.2, the permission backend added a ``metadata``
+        table. If the permission and storage backends share
+        credentials, this will be the same table created by the
+        storage backend. This means either backend could create the
+        table without the knowledge of the other one. For this reason,
+        be careful to handle the case where the metadata table exists
+        but no version exists.
+
+        """
+        query = "SELECT tablename FROM pg_tables WHERE tablename = 'metadata';"
+        with self.client.connect() as conn:
+            result = conn.execute(query)
+            table_exists = result.rowcount > 0
+
+        if table_exists:
+            query = """
+            SELECT value AS version
+              FROM metadata
+             WHERE name = 'permission_schema_version'
+             ORDER BY LPAD(value, 3, '0') DESC
+             LIMIT 1;
+            """
+            with self.client.connect() as conn:
+                result = conn.execute(query)
+                if result.rowcount > 0:
+                    return int(result.fetchone()['version'])
+
```

Comment:
Indent this whole block one to the right and avoid the second connection creation.